backend/src/v2/compiler/argocompiler/container.go

```diff
@@ -527,38 +536,61 @@ func (c *workflowCompiler) getTaskRetryPolicySpec(name string) *pipelinespec.Pip
 	return taskSpec.RetryPolicy
 }
 
-func (c *workflowCompiler) getTaskRetryStrategy(name string) *wfapi.RetryStrategy {
+func (c *workflowCompiler) getTaskRetryParameters(name string) *[]wfapi.Parameter {
 	retryPolicy := c.getTaskRetryPolicySpec(name)
 	if retryPolicy == nil {
-		return nil
-	}
+		return &[]wfapi.Parameter{}
+	}
+	// Create slice of parameters with "Name" field set.
+	parameters := []wfapi.Parameter{
+		{Name: paramRetryMaxCount},
+		{Name: paramRetryBackOffDuration},
+		{Name: paramRetryBackOffFactor},
+		{Name: paramRetryBackOffMaxDuration}}
+	return &parameters
+}
 
-	argoBackOffDuration := "0"
-	backoffDuration := retryPolicy.GetBackoffDuration()
-	if backoffDuration != nil {
-		argoBackOffDuration = strconv.FormatInt(backoffDuration.Seconds, 10)
-	}
+func (c *workflowCompiler) getTaskRetryParametersWithValues(name string) *[]wfapi.Parameter {
+	retryPolicy := c.getTaskRetryPolicySpec(name)
+	if retryPolicy == nil {
+		return &[]wfapi.Parameter{}
+	}
+	// Create slice of all retry inputs containing non-nil values.
+	parameters := append([]wfapi.Parameter{},
+		wfapi.Parameter{Name: paramRetryMaxCount,
+			Value: wfapi.AnyStringPtr(retryPolicy.MaxRetryCount)})
+	if retryPolicy.GetBackoffDuration() != nil {
+		retryBackOffDuration := wfapi.AnyStringPtr(retryPolicy.GetBackoffDuration().Seconds)
+		parameters = append(parameters,
+			wfapi.Parameter{Name: paramRetryBackOffDuration,
+				Value: retryBackOffDuration})
+	}
+	parameters = append(parameters,
+		wfapi.Parameter{Name: paramRetryBackOffFactor,
+			Value: wfapi.AnyStringPtr(retryPolicy.GetBackoffFactor())})
+	if retryPolicy.GetBackoffMaxDuration() != nil {
+		retryBackOffMaxDuration := wfapi.AnyStringPtr(retryPolicy.GetBackoffMaxDuration().Seconds)
+		parameters = append(parameters,
+			wfapi.Parameter{Name: paramRetryBackOffMaxDuration,
+				Value: retryBackOffMaxDuration})
+	}
+	return &parameters
+}
 
-	var argoMaxDuration string
-	backoffMaxDuration := retryPolicy.GetBackoffMaxDuration()
-	if backoffMaxDuration != nil {
-		argoMaxDuration = strconv.FormatInt(backoffMaxDuration.Seconds, 10)
-	}
-	backoff := &wfapi.Backoff{
-		Factor: &intstr.IntOrString{
-			Type:   intstr.Int,
-			IntVal: int32(retryPolicy.GetBackoffFactor()),
-		},
-		MaxDuration: argoMaxDuration,
-		Duration:    argoBackOffDuration,
+func (c *workflowCompiler) addParameterDefault(parameters []wfapi.Parameter, defaultValue string) *[]wfapi.Parameter {
```

Comment:
No need to return a pointer here.
```suggestion
func (c *workflowCompiler) addParameterDefault(parameters []wfapi.Parameter, defaultValue string) []wfapi.Parameter {
```

Reply:
Updated